sklearn/utils/estimator_checks.py

```diff
@@ -346,6 +347,27 @@ def check_estimators_dtypes(name, Estimator):
                 pass
 
 
+def check_estimators_empty_data_messages(name, Estimator, multi_output=False):
+    e = Estimator()
+    set_fast_parameters(e)
+    set_random_state(e, 1)
+
+    X_zero_samples = np.empty(0).reshape(0, 3)
+    # The precise message can change depending on whether X or y is
+    # validated first. Let us test the type of exception only:
+    assert_raises(ValueError, e.fit, X_zero_samples, [])
+
+    X_zero_features = np.empty(0).reshape(3, 0)
+    # the following y should be accepted by both classifiers and regressors
+    # and ignored by unsupervised models
+    if multi_output:
```

Comment:
So here you could do
``y = multioutput_estimator_convert_y_2d(name, y)``
And remove the multioutput argument. This is how it is done in all the other tests.